frontend/javascripts/oxalis/geometries/materials/plane_material_factory.ts

```diff
@@ -780,24 +855,36 @@ class PlaneMaterialFactory {
       isOrthogonal: this.isOrthogonal,
       lookupTextureWidth,
     });
+    return [
+      code,
+      { availableLayerIndexToGlobalLayerIndex: { value: availableLayerIndexToGlobalLayerIndex } },
+    ];
   }
 
   getVertexShader(): string {
-    return `
-precision highp float;
-
-varying vec4 worldCoord;
-varying vec4 modelCoord;
-varying vec2 vUv;
-varying mat4 savedModelMatrix;
-
-void main() {
-  vUv = uv;
-  modelCoord = vec4(position, 1.0);
-  savedModelMatrix = modelMatrix;
-  worldCoord = modelMatrix * vec4(position, 1.0);
-  gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
-}`;
+    const { initializedGpuFactor, maximumLayerCountToRender } =
+      Store.getState().temporaryConfiguration.gpuSetup;
+    // Don't compile code for segmentation in arbitrary mode
+    const [colorLayerNames, segmentationLayerNames, globalLayerCount] =
+      this.getLayersToRender(maximumLayerCountToRender);
+
+    const packingDegreeLookup = getPackingDegreeLookup();
+    const { dataset } = Store.getState();
+    const datasetScale = dataset.dataSource.scale;
+    const lookupTextureWidth = getLookupBufferSize(initializedGpuFactor);
+
+    return getMainVertexShader({
+      globalLayerCount,
+      colorLayerNames,
+      segmentationLayerNames,
+      packingDegreeLookup,
+      // Todo: this is not computed per layer. See #4018
+      dataTextureCountPerLayer: Model.maximumTextureCountForLayer,
```

Comment:
In https://github.com/scalableminds/webknossos/issues/4018 you mention that this could improve performance if it was "fixed". Just wanted to mention it in case you had forgotten about it (like me), no need to do it in this PR.